testing/union_dataset_test.py

```diff
@@ -0,0 +1,72 @@
+#
+# union_dataset_test.py
+# Francois-Michel L'Heureux, 2016-09-20
+# This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+#
+
+mldb = mldb_wrapper.wrap(mldb)  # noqa
+
+class UnionDatasetTest(MldbUnitTest):  # noqa
+
+    @classmethod
+    def setUpClass(cls):
+        ds = mldb.create_dataset({'id' : 'ds1', 'type' : 'sparse.mutable'})
```

Comment:
I would test with at least two rows and test with dataset with matching column name.  Also, if we expect the result to be equivalent to a JOIN than I would also test that.